src/main/java/controller/CreateNodeController.java

```diff
@@ -143,84 +150,112 @@ void mapClickedHandler(MouseEvent e) {
+        switch (clickedBtn.getText()) {
+            case "Floor 3":
+                setMapFloor("3");
+                if(stateIterator == 0) {node_floor = "3";}
+                else if(stateIterator == 2) {showAllNodes("3");}
+                break;
+            case "Floor 2":
+                setMapFloor("2");
+                if(stateIterator == 0) {node_floor = "2";}
+                else if(stateIterator == 2) {showAllNodes("2");}
+                break;
+            case "Floor 1":
+                setMapFloor("1");
+                if(stateIterator == 0) {node_floor = "1";}
+                else if(stateIterator == 2) {showAllNodes("1");}
+                break;
+            case "L1":
+                setMapFloor("L1");
+                if(stateIterator == 0) {node_floor = "L1";}
+                else if(stateIterator == 2) {showAllNodes("L1");}
+                break;
+            case "L2":
+                setMapFloor("L2");
+                if(stateIterator == 0) {node_floor = "L2";}
+                else if(stateIterator == 2) {showAllNodes("L2");}
+                break;
+            case "Ground":
+                setMapFloor("G");
+                if(stateIterator == 0) {node_floor = "G";}
+                else if(stateIterator == 2) {showAllNodes("G");}
+                break;
+            default:
+                System.out.println("WHAT BUTTON WAS PRESSED?????");
+                break;
+        }
+    }
+
     // This function is bound to the generated node circles
     // It will run every time you click a node to add as edge
     @FXML
     void handleNodeClicked(MouseEvent e, Node selectedNode) {
         Circle nodeCircle = (Circle) e.getSource();
         // find if the node was already clicked
         if (nodeCircle.getFill().equals(DEFAULT_NODE_COLOR)) {
-            // add node to edge list
-            edges.add(new Edge(myCreatedNode, selectedNode));
+            // add to selected node list
+            chosenEdgeNodes.add(selectedNode);
             // change color
             nodeCircle.setFill(SELECTED_NODE_COLOR);
-            // draw line path
-            appendLine(selectedNode);
         } else {
             // already selected
             // change color
             nodeCircle.setFill(DEFAULT_NODE_COLOR);
-            // remove this as an edge
-            ArrayList<Edge> updatedEdges = new ArrayList<Edge>();
-            // filter through edges
-            for (Edge edge : edges) {
-                if (!edge.getNode2().getNodeID().equals(selectedNode.getNodeID())) {
-                    updatedEdges.add(edge);
-                }
-            }
-            edges = updatedEdges;
-            // remove the line
-            filterLine(selectedNode);
+            // remove from selected node list
+            chosenEdgeNodes.remove(selectedNode);
         }
         checkEnoughEdges();
     }
 
+    private void setMapFloor(String floor) throws IOException {
```

Comment:
One of my optimizations is to create a static hashmap of the images so you don't have to read from disk each time